hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodiePositionBasedFileGroupRecordBuffer.java

```diff
@@ -159,6 +167,10 @@ public boolean hasNext() throws IOException {
     while (baseFileIterator.hasNext()) {
       T baseRecord = baseFileIterator.next();
       Pair<Option<T>, Map<String, Object>> logRecordInfo = records.remove(nextRecordPosition++);
+      // A delete operation.
+      if (!logRecordInfo.getLeft().isPresent()) {
```

Comment:
Is this necessary? logRecordInfo can be null and deletes should be handled in the merger.

Reply:
Not necessary. But save some cpu cycles though. Let me remove it for now.